ctk/common/src/main/kotlin/org/codice/compliance/verification/profile/ProfilesVerifier.kt

```diff
@@ -31,11 +32,15 @@ class ProfilesVerifier(private val node: Node) {
      * Verify Error Response against the Profiles document.
      * This should be called explicitly if an error is expected.
      */
-    fun verifyErrorResponse() {
-        if (node.allChildren("Assertion").isNotEmpty())
-            throw SAMLComplianceException.create(SAMLProfiles_4_1_4_2_l,
+    fun verifyErrorResponseAssertion(samlErrorCode: SAMLSpecRefMessage? = null) {
+        if (node.allChildren("Assertion").isNotEmpty()) {
+            var exceptions: Array<SAMLSpecRefMessage> = arrayOf(SAMLProfiles_4_1_4_2_l)
```

Comment:
✏️ How about 
```kotlin
val exceptions: Array<SAMLSpecRefMessage> = 
  if (samlErrorCode == null) arrayOf(SAMLProfiles_4_1_4_2_l)
 else arrayOf(samlErrorCode, SAMLProfiles_4_1_4_2_l)
```
Could probably use a `var` and do a conditional append as well.

Reply:
👍 I like that